### PR TITLE
Prepare for upcoming change to HttpRequest and HttpClientResponse

### PR DESCRIPTION
**Pull request by tvolkert**
An upcoming change to the Dart SDK will change `HttpRequest` and
`HttpClientResponse` from implementing `Stream<List<int>>` to
implementing `Stream<Uint8List>`.

This forwards-compatible change prepares for that SDK breaking
change by casting the Stream to `List<int>` before transforming
it.

https://github.com/dart-lang/sdk/issues/36900

### CODE REVIEW

**Comment by tvolkert** [comment]
@kevmoo

**Comment by davidmorgan** [comment]
/subscribe

This change is needed for the google3 SDK roll. Could you please let me know when it's in google3?

**Comment by davidmorgan** [comment]
My mistake, what I need for the roll is #57

**Comment by davidmorgan** [comment]
Sorry, actually I don't think either of these are sufficient; here's the stack trace I'm seeing:

  Test failed: Caught type 'Utf8Decoder' is not a subtype of type 'StreamTransformer<Uint8List, String>' of 'streamTransformer'#0      Stream.transform (dart:async/stream.dart)
  #1      HttpBodyHandlerImpl.process.asText (package:http_server/src/http_body_impl.dart:92:12)
  #2      HttpBodyHandlerImpl.process (package:http_server/src/http_body_impl.dart:141:22)
  #3      HttpBodyHandlerImpl.processResponse (package:http_server/src/http_body_impl.dart:72:12)
  #4      HttpBodyHandler.processResponse (package:http_server/src/http_body.dart:125:32)

**Comment by tvolkert** [comment]
Ah, I missed that in my initial sweep.  I updated this PR accordingly.

**Review by whesse — APPROVED** [review]
LGTM

**Comment by whesse** [comment]
LGTM now modified by suggesting a that the code pattern
stream.transform(transformer)
be modified to
transformer.bind(stream)
instead of
steam.<List<int>>cast().transform(transformer).

**Comment by tvolkert** [comment]
@whesse done.

**Review by whesse — APPROVED** [review]
I'm not sure why Utf8Codec.decode(response.cast<List<int>>()) needs the cast, but LGTM.

**Comment by tvolkert** [comment]
> I'm not sure why Utf8Codec.decode(response.cast<List>()) needs the cast

It uses `Stream.transform()` in its impl.  I sent https://dart-review.googlesource.com/c/sdk/+/108161 to fix that 🙂

**Comment by davidmorgan** [comment]
Friendly ping; waiting for this to land in google3. Thanks :)

**Comment by tvolkert** [comment]
I don't have permission to land.  @whesse or @kevmoo could one of you land it?

**Comment by kevmoo** [comment]
@tvolkert – guessing you want this published, too?

**Comment by tvolkert** [comment]
Yes please 🙂

**Comment by kevmoo** [comment]
Done! and tagged

**Pull request merged**
